crates/hir-def/src/item_tree.rs

```diff
@@ -277,23 +276,23 @@ struct ItemVisibilities {
 
 #[derive(Default, Debug, Eq, PartialEq)]
 struct ItemTreeData {
-    uses: Arena<Use>,
-    extern_crates: Arena<ExternCrate>,
-    extern_blocks: Arena<ExternBlock>,
-    functions: Arena<Function>,
-    structs: Arena<Struct>,
-    unions: Arena<Union>,
-    enums: Arena<Enum>,
-    consts: Arena<Const>,
-    statics: Arena<Static>,
-    traits: Arena<Trait>,
-    trait_aliases: Arena<TraitAlias>,
-    impls: Arena<Impl>,
-    type_aliases: Arena<TypeAlias>,
-    mods: Arena<Mod>,
-    macro_calls: Arena<MacroCall>,
-    macro_rules: Arena<MacroRules>,
-    macro_defs: Arena<Macro2>,
+    uses: FxHashMap<ItemTreeAstId<Use>, Use>,
```

Comment:
HashMap is more memory heavy than `Arena`, and item trees consume a lot of memory, so I'm worried this will substantially increase our memory usage.

Reply:
Oh ye dw, this is only temporary. I am planning on collapsing all of these into one. I am just doing these changes one by one.